qlty-cli/src/commands/auth/login.rs

```diff
@@ -1,13 +1,41 @@
+use std::io::stdin;
+
 use crate::{Arguments, CommandError, CommandSuccess};
 use anyhow::Result;
 use clap::Args;
-use qlty_cloud::load_or_retrieve_auth_token;
+use console::style;
+use qlty_cloud::{load_or_retrieve_auth_token, store_auth_token};
 
 #[derive(Args, Debug)]
-pub struct Login {}
+pub struct Login {
+    /// Provide a CLI token manually or use "-" to read from standard input (see https://qlty.sh/user/settings/cli)
+    #[arg(long)]
+    pub token: Option<String>,
+}
 
 impl Login {
     pub fn execute(&self, _args: &Arguments) -> Result<CommandSuccess, CommandError> {
+        if let Some(mut token) = self.token.clone() {
+            if token == "" || token == "-" {
+                eprint!(
+                    "Generate a token from {} and paste it here.\nToken: ",
+                    style("https://qlty.sh/user/settings/cli")
+                        .underlined()
+                        .green()
+                );
+                let line = &mut String::new();
+                stdin().read_line(line)?;
```

Comment:
Rather than printing and reading input directly, we have been using the dialoguer crate, which I think would make sense to use here for consistency:

https://docs.rs/dialoguer/latest/dialoguer/struct.Input.html